src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        if (action.extraData && typeof action.extraData === 'object') {
```

Comment:
The check `typeof action.extraData === 'object'` should also ensure that `action.extraData` is not null, as `typeof null` is also 'object'. Consider using `action.extraData && typeof action.extraData === 'object' && !Array.isArray(action.extraData)` to ensure it's a non-null object.